freeipa-client/src/main/java/com/sequenceiq/freeipa/client/FreeIpaClient.java

```diff
@@ -690,4 +692,54 @@ public void checkIfClientStillUsable(FreeIpaClientException e) throws FreeIpaCli
         return Map.of("__dns_name__", name);
     }
 
+    public Set<SudoCommand> sudoCommandFindAll() throws FreeIpaClientException {
+        List<Object> flags = List.of();
+        Map<String, Object> params = Map.of(
+                "sizelimit", 0,
+                "timelimit", 0
+        );
+        ParameterizedType type = TypeUtils
+                .parameterize(Set.class, SudoCommand.class);
+        return (Set<SudoCommand>) invoke("sudocmd_find", flags, params, type).getResult();
+    }
+
+    public SudoCommand sudoCommandAdd(String sudoCommand) throws FreeIpaClientException {
+        List<Object> flags = List.of(sudoCommand);
+        Map<String, Object> params = Map.of();
+        return (SudoCommand) invoke("sudocmd_add", flags, params, SudoCommand.class).getResult();
+    }
+
+    public SudoRule sudoRuleShow(String ruleName) throws FreeIpaClientException {
+        List<Object> flags = List.of(ruleName);
+        Map<String, Object> params = Map.of();
+        return (SudoRule) invoke("sudorule_show", flags, params, SudoRule.class).getResult();
+    }
+
+    public SudoRule sudoRuleAdd(String sudoRuleName, boolean hostCategoryAll) throws FreeIpaClientException {
+        List<Object> flags = List.of(sudoRuleName);
+        Map<String, Object> params = new HashMap<>();
+        if (hostCategoryAll) {
+            params.put("hostcategory", "all");
+        }
+
+        return (SudoRule) invoke("sudorule_add", flags, params, SudoRule.class).getResult();
+    }
+
+    public SudoRule sudoRuleAddAllowCommand(String ruleName, String sudoCommand) throws FreeIpaClientException {
+        List<Object> flags = List.of(ruleName);
+        Map<String, Object> params = Map.of("sudocmd", sudoCommand);
+        return (SudoRule) invoke("sudorule_add_allow_command", flags, params, SudoRule.class).getResult();
+    }
+
+    public SudoRule sudoRuleAddDenyCommand(String ruleName, String sudoCommand) throws FreeIpaClientException {
+        List<Object> flags = List.of(ruleName);
+        Map<String, Object> params = Map.of("sudocmd", sudoCommand);
+        return (SudoRule) invoke("sudorule_add_deny_command", flags, params, SudoRule.class).getResult();
+    }
+
+    public SudoRule sudoRuleAddGroup(String ruleName, String group) throws FreeIpaClientException {
+        List<Object> flags = List.of(ruleName);
+        Map<String, Object> params = Map.of("group", group);
+        return (SudoRule) invoke("sudorule_add_user", flags, params, SudoRule.class).getResult();
+    }
```

Comment:
could you implement these as a subclass of `AbstractFreeipaOperation`?

Reply:
Done with one exception: sudoCommandFindAll
The AbstractFreeipaOperation is not prepared for collections so if I'd use a Set as a generic parem, I should use `Optional<Set<>>` as a return type of `AbstractFreeipaOperation.invoke(FreeIpaClient freeIpaClient)` what I would not like to see in the codebase... I hope you can agree :-) Refactoring all the related classes is out of this PR's scope. If you think we should handle it, it needs a separated ticket to handle.

Reply:
yeah thats fine, we will figure that one out later. could you remove these methods from the client which has been replaced by operations?

Reply:
TBH, I would not as the usage of the requested AbstractFreeipaOperation will make the unit testing of SudoRuleService a nightmare. Can you accept to keep these methods in FreeIpaClient so we can simply mock FreeIpaClient itself without having RPCResponse magics in SudoRuleService?

Reply:
hmm, fine, thats a good point